### PR TITLE
Process input handler refactor

### PR DESCRIPTION
**Pull request by zrock-dev**
# Task-207 Implementation of Input intermediate

https://tree.taiga.io/project/joseluis-teran-coffeetime/task/207

## Description
Implementation of : https://github.com/Pending-Name-21/arquitecture/pull/20

## Type of Change

Put an `x` in all the boxes that apply:

- [ ] ✨ New Feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug Fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking Change (fix or feature that would cause existing functionality to change)
- [x] 🧹 Code Refactor
- [ ] ✅ Build Configuration Change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

## Pull Request Checklist

Put an `x` in all the boxes that apply:

- [x] My commit messages are detailed.
- [x] My code follows the code style of this project.
- [ ] No existing features have been broken without good reason.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests pass.

### CODE REVIEW

File: bridge/src/main/java/com/bridge/processinputhandler/AEventManager.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package com.bridge.processinputhandler;
+
```
Comment (codeFactory12) [inline bridge/src/main/java/com/bridge/processinputhandler/AEventManager.java:2]: src/main/java/com/bridge/processinputhandler/AEventManager.java
          @@ -1,7 +1,6 @@
           package·com.bridge.processinputhandler;
           
           import·com.bridge.ipc.IEventBuffer;
          -
           import·java.util.ArrayList;
           import·java.util.List;
           import·java.util.concurrent.ConcurrentLinkedQueue;
Doesnt acomlish with format, review with ./gradlew build

Run './gradlew :bridge:spotlessApply'

**Review by codeFactory12 — CHANGES_REQUESTED** [review]

File: bridge/src/main/java/com/bridge/processinputhandler/AEventManager.java (hunk 1 above)
Comment (codeFactory12) [reply bridge/src/main/java/com/bridge/processinputhandler/AEventManager.java:2]: There are more archieves that needs a format fix such us:
src/test/java/com/bridge/ipc/ReceiverTest.java
src/test/java/com/bridge/processinputhandler/AEventManagerTest.java
src/test/java/com/bridge/processinputhandler/InputVerifierTest.java
src/test/java/com/bridge/GameTest.java
src/test/java/com/bridge/EventGenerator.java
src/main/java/com/bridge/processinputhandler/IPublisher.java
can be fixed using : **./gradlew :bridge:spotlessApply**

File: bridge/src/test/java/com/bridge/GameTest.java [hunk 2]
```diff
@@ -1,18 +1,22 @@
...
-import static org.junit.jupiter.api.Assertions.assertTrue;
-import static org.junit.jupiter.api.Assertions.fail;
-
 import com.bridge.core.exceptions.GameException;
 import com.bridge.core.exceptions.initializerhandler.NotPossibleToInitializeSubscribersException;
 import com.bridge.gamesettings.AGameSettings;
 import com.bridge.initializerhandler.GameInitializer;
 import com.bridge.processinputhandler.InputVerifier;
+import com.bridge.processinputhandler.KeyboardEventManager;
+import com.bridge.processinputhandler.MouseEventManager;
```
Comment (codeFactory12) [inline bridge/src/test/java/com/bridge/GameTest.java:12]: There is no MouseEventManager uploaded

File: bridge/src/main/java/com/bridge/processinputhandler/KeyboardEventManager.java [hunk 3]
```diff
@@ -26,26 +24,23 @@
...
  */
-public class KeyboardEventPublisher implements IPublisher {
-    private final ConcurrentLinkedQueue<KeyboardEvent> buffer;
-    private final List<IKeyboardEventSubscriber> subscribers;
+public class KeyboardEventManager extends AEventManager<Keyboard> {
 
     /**
      * Constructs a KeyboardEventPublisher with the specified event buffer.
      *
      * @param buffer the concurrent linked queue used to store keyboard events
```
Comment (codeFactory12) [inline bridge/src/main/java/com/bridge/processinputhandler/KeyboardEventManager.java]: Cannot resolve symbol 'buffer' on line 32

**Review by codeFactory12 — CHANGES_REQUESTED** [review]
There are errors in tests clases that are not uploeaded sich us 
test/java/com/bidge
ipc/SocketServerTest
Expected 0 arguments but found 1 on line 13
processinputhandler/EventTypeTest
Cannot resolve symbol 'EventType' on line 12
GameWithExceptionsTest
Cannot resolve symbol 'Utils'

**Review by codeFactory12 — APPROVED** [review]
Nice joob

**Review by fabianromeroclaros — APPROVED** [review]

**Pull request merged**